program-tests/pre-authorized-debit-v1/init-smart-delegate.test.2.ts

```diff
@@ -0,0 +1,120 @@
+    await expect(
+      program.methods
+        .initSmartDelegate()
+        .accounts({
+          payer: payer.publicKey,
+          smartDelegate: invalidSmartDelegate,
+          systemProgram: SystemProgram.programId,
+        })
+        .signers([payer])
+        .rpc(),
+    ).to.eventually.to.be.rejectedWith(
+      /AnchorError caused by account: smart_delegate. Error Code: ConstraintSeeds. Error Number: 2006. Error Message: A seeds constraint was violated./,
+    );
+  });
+
+  it("should throw an error with an invalid system program id", async () => {
+    await expect(
+      program.methods
+        .initSmartDelegate()
+        .accounts({
+          payer: payer.publicKey,
+          smartDelegate: canonicalSmartDelegatePublicKey,
+          systemProgram: TOKEN_2022_PROGRAM_ID,
+        })
+        .signers([payer])
+        .rpc(),
+    ).to.eventually.to.be.rejectedWith(
+      /AnchorError caused by account: system_program. Error Code: InvalidProgramId. Error Number: 3008. Error Message: Program ID was not as expected./,
+    );
+  });
+
+  it("should create a smart delegate", async () => {
+    const [payerAccountInfoBefore] =
+      await provider.connection.getMultipleAccountsInfo([payer.publicKey]);
+    assert(payerAccountInfoBefore);
+
+    const signature = await program.methods
+      .initSmartDelegate()
+      .accounts({
+        payer: payer.publicKey,
+        smartDelegate: canonicalSmartDelegatePublicKey,
+        systemProgram: SystemProgram.programId,
+      })
+      .signers([payer])
+      .rpc();
+    const tx = await waitForTxToConfirm(signature, provider.connection);
+    assert(tx.meta?.logMessages);
+
+    // verify events
+    const eventGenerator = eventParser.parseLogs(tx.meta.logMessages);
+    const events = [...eventGenerator];
+    expect(events.length).to.equal(1);
+    const [smartDelegateEvent] = events;
+    expect(smartDelegateEvent).to.not.equal(null);
+    expect(smartDelegateEvent.name).to.equal("SmartDelegateInitialized");
+    expect(Object.keys(smartDelegateEvent.data).length).to.equal(2);
+    expect(smartDelegateEvent.data.payer!.toString()).to.equal(
+      payer.publicKey.toString(),
+    );
+    expect(smartDelegateEvent.data.smartDelegate!.toString()).to.equal(
+      canonicalSmartDelegatePublicKey.toString(),
+    );
+
+    // verify sol balances
+    const [payerAccountInfoAfter] =
+      await provider.connection.getMultipleAccountsInfo([payer.publicKey]);
+    assert(payerAccountInfoAfter);
+    expect(payerAccountInfoAfter.lamports).to.be.lessThan(
+      payerAccountInfoBefore.lamports,
+    );
+
+    // verify smart delegate account
+    const smartDelegateAccount = await program.account.smartDelegate.fetch(
+      canonicalSmartDelegatePublicKey,
+    );
+    expect(smartDelegateAccount.bump).to.equal(canonicalSmartDelegateBump);
+  });
+
+  it("should fail if trying to init a smart delegate twice", async () => {
```

Comment:
A bit of an anti-pattern to have tests rely on other tests.
This test relies on the previous test to pass. 

But in the grand scheme of things it doesn't matter as all tests in this describe block run sequentially and its a simple test file.

Reply:
Yeah in this case, I think it's fine tbh